File: cpp/0491递增子序列.cpp
#include<vector>
using namespace std;

vector<vector<int>> res;
vector<int> path;
void BackTracking(const vector<int> &nums, int startIndex)
{
    if (path.size() > 1)
    {
        res.push_back(path);
    }
    int used[201] = {0}; 
    for (int i = startIndex; i < nums.size(); i++)
    {
        if ((!path.empty() && nums[i] < path.back()) || used[nums[i] + 100] == 1)
        {
            continue;
        }
        used[nums[i] + 100] = 1; 
        path.push_back(nums[i]);
        BackTracking(nums, i + 1);
        path.pop_back();
    }
}

vector<vector<int>> findSubsequences(vector<int> &nums)
{
    
    BackTracking(nums, 0);

    return res;
}

File: python/main/0491递增子序列.py

def findSubsequences(nums: list) -> list:
    res=[]
    temp=[]
    
    def _backTracking(start:int=0):
        nonlocal nums
        memo=[0]*201

        if(len(temp)>1):
            res.append(temp.copy())

        for i,v in enumerate(nums[start:],start):

            if(memo[v+100] or (len(temp) and temp[-1]>v)):
                continue
            
            temp.append(v)
            memo[v+100]=1
            _backTracking(i+1)
            temp.pop()
    
    _backTracking()
    return res